classes/03class/exercises/c03-docker01/gepifanio/Dockerfile

```diff
@@ -0,0 +1,6 @@
+FROM alpine
+ENV TERRAFORM_VERSION=0.12.20
```

Comment:
We usually use `ENV` when we would like to change the variable after the container is built and to be available for our application at running time. For instance, a feature flag for our application. Or, some API key which our app is using and that can be modified without building a new image.

We don't need those features for this scenario. So can you think in the alternative way to define variables?

> **Extra:** that alternative way has a bonus that can be passed at building time (which is not true for ENV variables). So we would be able to build several base terraform images with different terraform version using the same unchanged Dockerfile. 🤯

Reply:
![image](https://user-images.githubusercontent.com/16002934/87519874-8aef0a80-c6c5-11ea-9577-fed0c8bc83a4.png)

Nice thanks for this insight, I will be using ARG in this case